jhotdraw-core/src/main/java/org/jhotdraw/draw/DefaultDrawing.java

```diff
@@ -146,35 +146,36 @@ public Figure findFigureBehind(Point2D.Double p, Figure figure) {
   }
 
   @Override
-  public Figure findFigureBehind(Point2D.Double p, double scaleDenominator, Figure figure) {
-    boolean isBehind = false;
-    for (Figure f : getFiguresFrontToBack()) {
-      if (isBehind) {
-        if (f.isVisible() && f.contains(p, scaleDenominator)) {
-          return f;
+  public Figure findFigureBehind(Point2D.Double p, Object obj) {
```

Comment:
The method findFigureBehind(Point2D.Double p, Object obj) is missing a mechanism to handle the case where neither of the conditions (if, else if) is satisfied. One of the following actions needs to be taken:
1) Add a return statement, or
2) Throw an exception in case none of the branches is exercised